### ISSUE TITLE
Numerical Instabilities

### ISSUE THREAD

**Issue by cooperoelrichs** [cooperoelrichs/Discrete_Choice#4]
Overflows occur in the exponential.

This might be fixable using a method similar to scipy.misc.logsumexp()

**Comment by cooperoelrichs** [comment]
The exponential operation (e ^ x) is very susceptible floating point overflows, and this creates problems when trying to estimate logit models, which rely on exponentials for calculating probabilities.
1. Simple to fix when using Softmax (Multinomial Logit)
   
   ```
   V = utilities_matrix
   e_V = exp(V - mean(V, axis=1))
   P = e_V / sum(e_V, axis=1)
   ```
2. But how do we fix this when using Nested Logit?
   - Adjust V before calculating exp
     ![image](https://cloud.githubusercontent.com/assets/8163246/10498771/c9f40050-730f-11e5-8a2a-f000584108cc.png)
   - Scale X values that are input to the Nested Logit Estimator.
   
   ```
   v = bias + beta_1 * x_1 + beta_2 * x_2 + ...
   beta_1 * x_1 = beta_1' * (x_1 / scale)
   beta_1 = beta_1' / scale
   ```

**Comment by cooperoelrichs** [comment]
Try scaling X before estimating
1. Compare output accuracy
2. Compare predicted probabilities

**Comment by cooperoelrichs** [comment]
We will adjust V before calculating the exponential. As is shown above, P is only dependant on the relative difference between the V values.
